presto-main/src/main/java/com/facebook/presto/tdigest/TDigest.java

```diff
@@ -566,6 +566,60 @@ else if (activeCentroids == 1) {
         return weightedAverage(mean[n - 1], z1, max, z2);
     }
 
+    public double trimmedMean(double l, double u)
+    {
+        checkArgument(l >= 0 && l <= 1, "l should be in [0,1], got %s", l);
+        checkArgument(u >= 0 && u <= 1, "u should be in [0,1], got %s", u);
+
+        if (unmergedWeight > 0) {
+            compress();
+        }
+
+        if (activeCentroids == 0 || l >= u) {
+            return Double.NaN;
+        }
+
+        if (l == 0 && u == 1) {
+            return sum / totalWeight;
+        }
+
+        double lowerIndex = l * totalWeight;
+        double upperIndex = u * totalWeight;
+
+        int n = activeCentroids;
+
+        double weightSoFar = 0;
+        double sumInBounds = 0;
+        double weightInBounds = 0;
+        for (int i = 0; i < n; i++) {
+            if (weightSoFar < lowerIndex && lowerIndex <= weightSoFar + weight[i] && upperIndex <= weightSoFar + weight[i]) {
```

Comment:
for readability I would recommend getting rid of the comments and refactoring the checks to a separate method with meaningful names.

e.g.
```
if (isInSameWeightInterval(..))
```

Reply:
I'm not sure if that would improve readability since we'd need to pass multiple arguments (weightSoFar, lowerIndex, upperIndex, weight[i]) and the function would only be used once. Also it would be inconsistent with `getQuantile` that uses comments to explain each case